### PR TITLE
Improvement: parser and render

### PR DESCRIPTION
**Pull request by Zorin95670**
Improve render and parser after testing on leto-modelizer with read data.

### CODE REVIEW

File: src/assets/metadata/generic.json [hunk 1]
```diff
@@ -1,8 +1,11 @@
 {
   "workflow": {
     "type": "workflow",
-    "model": "DefaultContainer",
+    "model": "workflow",
     "icon": "DefaultWorkflow",
+    "displayName": "Workflow",
+    "description": "A workflow is a configurable automated process that will run one or more jobs.<br/><br/>Workflows are defined by a YAML file checked in to your repository and will run when triggered by an event in your repository, or they can be triggered manually, or at a defined schedule.",
```
Comment (k-y0u) [inline src/assets/metadata/generic.json:7]: `and will run` : what will run ?

Comment (k-y0u) [reply src/assets/metadata/generic.json:7]: sentence too long.

File: src/assets/metadata/generic.json [hunk 2]
```diff
@@ -70,32 +92,34 @@
...
     "type": "step",
-    "model": "DefaultModel",
+    "model": "step",
     "icon": "DefaultStep",
-    "available": true,
+    "available": false,
     "parentTypes": [
       "job"
     ],
+    "description": "Task of the job. Steps can run commands, run setup tasks, or run an action in your repository, a public repository, or an action published in a Docker registry.",
```
Comment (k-y0u) [inline src/assets/metadata/generic.json:101]: sentence too long

File: src/assets/metadata/step.json [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
+  "displayName": "Reusable step",
   "icon": "ReusableStep",
   "name": "reusable-step",
   "attributes": [{
     "name": "uses",
-    "type": "String"
+    "type": "String",
+    "required": true,
+    "displayName": "External action",
+    "description": "Selects an action to run as part of a step in your job. An action is a reusable unit of code. You can use an action defined in the same repository as the workflow, a public repository, or in a published Docker container image.",
```
Comment (k-y0u) [inline src/assets/metadata/step.json:21]: Select

File: src/assets/metadata/trigger.json [hunk 4]
```diff
@@ -1,17 +1,27 @@
...
-    "type": "Array"
+    "type": "Array",
+    "description": "Select branch name on push event.",
+    "url": "https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs"
   }]
 }, {
   "type": "trigger",
   "icon": "PullRequestTrigger",
+  "displayName": "Trigger - pull request",
+  "description": "Workflow triggers are events that cause a workflow to run.<br/><br/>This event with trigger workflow on a pull request event.",
```
Comment (k-y0u) [inline src/assets/metadata/trigger.json:18]: Workflow trigger is an event

Comment (k-y0u) [reply src/assets/metadata/trigger.json:18]: `This event with trigger workflow on a pull request event.`
where is the verb ?

File: src/assets/metadata/trigger.json [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
   "action": "pull_request",
   "attributes": [{
     "name": "types",
-    "type": "Array"
+    "type": "Array",
+    "displayName": "Activity types",
+    "description": "Runs your workflow when activity on a pull request in the workflow's repository occurs.",
```
Comment (k-y0u) [inline src/assets/metadata/trigger.json:24]: when an activity

File: tests/resources/metadata/default.json [hunk 6]
```diff
@@ -115,104 +139,30 @@
...
     "available": true
   },
   {
     "type": "trigger",
     "icon": "PushTrigger",
-    "model": null,
-    "parentTypes": [],
+    "model": "trigger",
+    "displayName": "Trigger - push",
+    "description": "Workflow triggers are events that cause a workflow to run.<br/><br/>This event with trigger workflow on a push event.",
```
Comment (k-y0u) [inline tests/resources/metadata/default.json:153]: Workflow trigger is an event

File: tests/resources/metadata/default.json [hunk 7]
```diff
@@ -248,18 +201,29 @@
...
     "action": "push"
   },
   {
     "type": "trigger",
     "icon": "PullRequestTrigger",
-    "model": null,
-    "parentTypes": [],
+    "model": "trigger",
+    "displayName": "Trigger - pull request",
+    "description": "Workflow triggers are events that cause a workflow to run.<br/><br/>This event with trigger workflow on a pull request event.",
```
Comment (k-y0u) [inline tests/resources/metadata/default.json:214]: same

File: tests/resources/metadata/default.json [hunk 8]
```diff
@@ -278,12 +242,15 @@
       {
         "name": "types",
         "type": "Array",
+        "displayName": "Activity types",
+        "description": "Runs your workflow when activity on a pull request in the workflow's repository occurs.",
```
Comment (k-y0u) [inline tests/resources/metadata/default.json:246]: Run

File: tests/resources/metadata/default.json [hunk 9]
```diff
@@ -295,36 +262,29 @@
...
     "action": "pull_request"
   },
   {
     "type": "step",
     "icon": "CommandStep",
-    "model": null,
-    "parentTypes": [],
+    "model": "step",
+    "displayName": "Command step",
+    "description": "Task of the job. Steps can run commands, run setup tasks, or run an action in your repository, a public repository, or an action published in a Docker registry.",
```
Comment (k-y0u) [inline tests/resources/metadata/default.json:275]: Step can run commands

File: tests/resources/metadata/default.json [hunk 10]
```diff
@@ -396,36 +365,29 @@
...
     "name": "command-step"
   },
   {
     "type": "step",
     "icon": "ReusableStep",
-    "model": null,
-    "parentTypes": [],
+    "model": "step",
+    "displayName": "Reusable step",
+    "description": "Task of the job. Steps can run commands, run setup tasks, or run an action in your repository, a public repository, or an action published in a Docker registry.",
```
Comment (k-y0u) [inline tests/resources/metadata/default.json:378]: same

File: tests/resources/metadata/default.json [hunk 11]
```diff
@@ -480,14 +448,17 @@
       {
         "name": "uses",
         "type": "String",
+        "displayName": "External action",
+        "description": "Selects an action to run as part of a step in your job. An action is a reusable unit of code. You can use an action defined in the same repository as the workflow, a public repository, or in a published Docker container image.",
```
Comment (k-y0u) [inline tests/resources/metadata/default.json:452]: select an action

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/assets/metadata/trigger.json (hunk 4 above)
Comment (Zorin95670) [reply src/assets/metadata/trigger.json:18]: done

File: src/assets/metadata/trigger.json (hunk 5 above)
Comment (Zorin95670) [reply src/assets/metadata/trigger.json:24]: done

File: tests/resources/metadata/default.json (hunk 6 above)
Comment (Zorin95670) [reply tests/resources/metadata/default.json:153]: done

File: tests/resources/metadata/default.json (hunk 7 above)
Comment (Zorin95670) [reply tests/resources/metadata/default.json:214]: done

File: src/assets/metadata/trigger.json [hunk 12]
```diff
@@ -1,17 +1,27 @@
...
-    "type": "Array"
+    "type": "Array",
+    "description": "Select branch name on push event.",
+    "url": "https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs"
   }]
 }, {
   "type": "trigger",
   "icon": "PullRequestTrigger",
+  "displayName": "Trigger - pull request",
+  "description": "Workflow trigger is an event that cause a workflow to run.<br/><br/>This event will trigger the workflow on a pull request event.",
```
Comment (justinehell) [inline src/assets/metadata/trigger.json:18]: `an event that cause` -> `an event that causes`

File: src/assets/metadata/trigger.json [hunk 13]
```diff
@@ -1,17 +1,27 @@
...
   "action": "pull_request",
   "attributes": [{
     "name": "types",
-    "type": "Array"
+    "type": "Array",
+    "displayName": "Activity types",
+    "description": "Run your workflow when an activity on a pull request in the workflow's repository occurs.",
```
Comment (justinehell) [inline src/assets/metadata/trigger.json:24]:
``` suggestion
    "description": "Run your workflow when an activity on a pull request occurs in the workflow's repository.",
```

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/assets/metadata/trigger.json (hunk 12 above)
Comment (Zorin95670) [reply src/assets/metadata/trigger.json:18]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**